### PR TITLE
enzo-cappa fix: add warn log when a file failed to be ingested

### PR DESCRIPTION
**Pull request by sfc-gh-rcheng**
Forked repos fail our merge gates because [we can't share github creds](https://github.com/snowflakedb/snowflake-kafka-connector/pull/550#issuecomment-1595192172) for integration and e2e tests. Creating a duplicate of @enzo-cappa 's PR https://github.com/snowflakedb/snowflake-kafka-connector/pull/550. Description from their PR below:

From https://github.com/snowflakedb/snowflake-kafka-connector/issues/549: Add a WARN level log when a file failed to be ingested according to the copy history.

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeIngestionServiceV1.java [hunk 1]
```diff
@@ -142,7 +142,14 @@ public Map<String, InternalUtils.IngestedFileStatus> readIngestReport(final List
...
             numOfRecords++;
 
-            fileStatus.put(file.getPath(), convertIngestStatus(file.getStatus()));
+            final InternalUtils.IngestedFileStatus ingestionStatus =
+                convertIngestStatus(file.getStatus());
+            fileStatus.put(file.getPath(), ingestionStatus);
+            // Log errors
+            if (InternalUtils.IngestedFileStatus.FAILED.equals(ingestionStatus)
+                || InternalUtils.IngestedFileStatus.PARTIALLY_LOADED.equals(ingestionStatus)) {
+              LOGGER.warn("Failed to load file {}", file.getPath());
```
Comment (sfc-gh-japatel) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeIngestionServiceV1.java:151]: lets also add pipeName here.

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm, minor comment about pipename

**Review by sfc-gh-asen — APPROVED** [review]

**Review by sfc-gh-tzhang — APPROVED** [review]
LGTM, please add a meaningful description instead of just linking the old PR

**Comment by sfc-gh-rcheng** [comment]
Added description to PR

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm! thanks

**Pull request merged**